### DISCUSSION TITLE
Pyinstaller and Pyside2 ersion `Qt_5.15' not found problem

### DISCUSSION THREAD

**Discussion by hobik** [pyinstaller/pyinstaller#6501] [Help]
I want to convert `onefile` mygui file  with pyinstaller. If I open the `mygui` with `python3 mygui.py` it's openning successfully. But when I convert to onefile with pyinstaller and trying to open ./mygui, I get below errors:

       Traceback (most recent call last):
          File "mygui.py", line 6, in <module>
            from PySide2.QtWidgets import QApplication
        ImportError: /tmp/_MEIlK1RGy/libQt5Core.so.5: version `Qt_5.15' not found (required by /tmp/_MEIlK1RGy/libpyside2.abi3.so.5.15)
        [20822] Failed to execute script 'mygui' due to unhandled exception!

I am using Ubuntu 20.04 how can I solve this problem ? Why the problem appears with executable file ?

I guess problem is because I am using Pyside2 on this project. My other projects (only using pyqt5) successfully converting executable  onefile and working correctly. How can I solve this problem I couldnt find any solution.  
I convert onefile with this command:

    pyinstaller --onefile  --windowed mygui.py

**EDIT**
I updated pyinstaller and this error replaced with below error:

```
Traceback (most recent call last):
  File "mygui.py", line 3, in <module>
AttributeError: module 'PySide2' has no attribute 'QtPrintSupport'
[16196] Failed to execute script 'mygui' due to unhandled exception!
```

**Comment by rokm** [comment]
How is `PySide2` installed on this system? Is it the version that is installed from the ubuntu's repository, or PyPI wheel installed via `pip install`?

This looks as if Qt libraries were collected from one source (e.g., system), while python extension(s) were collected from another. Can you check `build/<name>/Analysis-00.toc` file and see where `libQt5Core.so.5` was collected from, and where `libpyside2.abi3.so.5.15` was collected from?

**Comment by rokm** [comment]
> **EDIT** I updated pyinstaller and this error replaced with below error:
> 
> ```
> Traceback (most recent call last):
>   File "mygui.py", line 3, in <module>
> AttributeError: module 'PySide2' has no attribute 'QtPrintSupport'
> [16196] Failed to execute script 'mygui' due to unhandled exception!
> ```

Try adding `PySide2.QtPrintSupport` to hidden imports (e.g., add `--hiddenimport PySide2.QtPrintSupport` to your PyInstaller command.

**Reply by rokm** [reply]
Ah, nevermind, that probably won't make a difference, since you are already doing a direct import of this module in your script. I guess the question remains the same - is this a system-provided PySide2 or pip-installed one?
